File: CS/MarkerAndCheckbox/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraCharts;

namespace MarkerAndCheckbox {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            chart.DataSource = CreateChartData();

            chart.SeriesDataMember = "Month";
            chart.SeriesTemplate.ArgumentDataMember = "Section";
            chart.SeriesTemplate.ValueDataMembers.AddRange(new string[] { "Value" });
            chart.SeriesTemplate.View = new StackedBarSeriesView();

        }

        private DataTable CreateChartData() {

            DataTable table = new DataTable("Table1");

            table.Columns.Add("Month", typeof(String));
            table.Columns.Add("Section", typeof(String));
            table.Columns.Add("Value", typeof(Int32));

            table.Rows.Add(new object[] { "Jan", "Section1", 10 });
            table.Rows.Add(new object[] { "Jan", "Section2", 20 });
            table.Rows.Add(new object[] { "Feb", "Section1", 20 });
            table.Rows.Add(new object[] { "Feb", "Section2", 30 });
            table.Rows.Add(new object[] { "March", "Section1", 15 });
            table.Rows.Add(new object[] { "March", "Section2", 25 });

            return table;
        }

        private void chart_CustomDrawSeries(object sender, CustomDrawSeriesEventArgs e) {
            Bitmap lmi = new Bitmap(33, 15);
            Graphics g = Graphics.FromImage(lmi);
            g.FillRectangle(new SolidBrush(e.SeriesDrawOptions.Color), new Rectangle(0, 0, 17, 15));
            g.DrawRectangle(new Pen(e.SeriesDrawOptions.Color), new Rectangle(18, 0, 14, 14));
            if (e.Series.CheckedInLegend) {
                g.DrawLines(new Pen(e.SeriesDrawOptions.Color, 2), new Point[] { new Point(21, 6), new Point(25, 10), new Point(29, 3) });
            }
            e.LegendMarkerImage = lmi;
        }
    }
}

File: VB/MarkerAndCheckbox/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraCharts

Namespace MarkerAndCheckbox

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            chart.DataSource = CreateChartData()
            chart.SeriesDataMember = "Month"
            chart.SeriesTemplate.ArgumentDataMember = "Section"
            chart.SeriesTemplate.ValueDataMembers.AddRange(New String() {"Value"})
            chart.SeriesTemplate.View = New StackedBarSeriesView()
        End Sub

        Private Function CreateChartData() As DataTable
            Dim table As DataTable = New DataTable("Table1")
            table.Columns.Add("Month", GetType(String))
            table.Columns.Add("Section", GetType(String))
            table.Columns.Add("Value", GetType(Integer))
            table.Rows.Add(New Object() {"Jan", "Section1", 10})
            table.Rows.Add(New Object() {"Jan", "Section2", 20})
            table.Rows.Add(New Object() {"Feb", "Section1", 20})
            table.Rows.Add(New Object() {"Feb", "Section2", 30})
            table.Rows.Add(New Object() {"March", "Section1", 15})
            table.Rows.Add(New Object() {"March", "Section2", 25})
            Return table
        End Function

        Private Sub chart_CustomDrawSeries(ByVal sender As Object, ByVal e As CustomDrawSeriesEventArgs)
            Dim lmi As Bitmap = New Bitmap(33, 15)
            Dim g As Graphics = Graphics.FromImage(lmi)
            g.FillRectangle(New SolidBrush(e.SeriesDrawOptions.Color), New Rectangle(0, 0, 17, 15))
            g.DrawRectangle(New Pen(e.SeriesDrawOptions.Color), New Rectangle(18, 0, 14, 14))
            If e.Series.CheckedInLegend Then
                g.DrawLines(New Pen(e.SeriesDrawOptions.Color, 2), New Point() {New Point(21, 6), New Point(25, 10), New Point(29, 3)})
            End If

            e.LegendMarkerImage = lmi
        End Sub
    End Class
End Namespace
